Dalam menentukan apakah customer akan tetap menggunakan product atau tidak maka perlu dilakukan pengukuran statistik pada data penggunaan jasa telekomunikasi:

1) Pada kolom tenure, monthly charge, dan total charge coba lakukan perhitungan rata-rata, median, modus, dan standar deviasi-nya. Berikan rangkuman singkat dari hasil perhitungan tiap kolomnya.

In [ ]:
import pandas as pd

In [ ]:
data = pd.read_csv('/content/WA_Fn-UseC_-Telco-Customer-Churn.csv')
data

customerID  gender  SeniorCitizen  ... MonthlyCharges TotalCharges  Churn
0     7590-VHVEG  Female              0  ...          29.85        29.85     No
1     5575-GNVDE    Male              0  ...          56.95       1889.5     No
2     3668-QPYBK    Male              0  ...          53.85       108.15    Yes
3     7795-CFOCW    Male              0  ...          42.30      1840.75     No
4     9237-HQITU  Female              0  ...          70.70       151.65    Yes
...          ...     ...            ...  ...            ...          ...    ...
7038  6840-RESVB    Male              0  ...          84.80       1990.5     No
7039  2234-XADUH  Female              0  ...         103.20       7362.9     No
7040  4801-JZAZL  Female              0  ...          29.60       346.45     No
7041  8361-LTMKD    Male              1  ...          74.40        306.6    Yes
7042  3186-AJIEK    Male              0  ...         105.65       6844.5     No

[7043 rows x 21 columns]

Perhitungan untuk mencari mean, median, dan modus menggunakan measures of central tendency. Kemudian untuk mencari standar deviasi digunakan measures of diversion.

In [ ]:
#Perhitungan pada kolom Tenure
mean_tenure = data['tenure'].mean()
print("The mean of Tenure : {:.2f}".format(round(mean_tenure,2)))

mode_tenure = data['tenure'].mode()
print("The mode of Tenure : {}".format(int(mode_tenure)))

median_tenure = data['tenure'].median()
print("The median of Tenure : {}".format(int(median_tenure)))

std_tenure = data['tenure'].std()
print("The average of Tenure : {:.2f}".format(std_tenure))

The mean of Tenure : 32.37
The mode of Tenure : 1
The median of Tenure : 29
The average of Tenure : 24.56


Hasil perhitungan pada kolom tenure didapatkan berupa nilai rata-rata sebesar : 32.37, modus : 1, median : 29, dan standard deviasi : 24.56

In [ ]:
#Perhitungan pada kolom Monthly Charges
mean_monthly = data['MonthlyCharges'].mean()
print("The mean of Monthly Charges: {:.2f}".format(round(mean_monthly,2)))

mode_monthly = data['MonthlyCharges'].mode()
print("The mode of Monthly Charges: {}".format(int(mode_monthly)))

median_monthly = data['MonthlyCharges'].median()
print("The median of Monthly Charges: {}".format(int(median_monthly)))

std_monthly = data['MonthlyCharges'].std()
print("The average of Monthly Charges: {:.2f}".format(std_monthly))

The mean of Monthly Charges: 64.76
The mode of Monthly Charges: 20
The median of Monthly Charges: 70
The average of Monthly Charges: 30.09


Hasil perhitungan pada kolom monthly charges didapatkan berupa nilai rata-rata sebesar : 63.76, modus : 20, median : 70, dan standard deviasi : 30.09

Pada kolom TotalCharges tipe datanya adalah object. Sehingga perlu diubah tipe data object menjadi numeric. Dan terdapat juga data yang Nan (null), sehingga dapat menggunakan errors='cource' untuk mengubah nilai null menjadi NaN.

In [ ]:
data['TotalCharges'] = pd.to_numeric(data['TotalCharges'], errors='coerce')
data.dtypes

customerID           object
gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges        float64
Churn                object
dtype: object

In [ ]:
#Perhitungan pada kolom Total Charges
mean_total = data['TotalCharges'].mean()
print("The mean of Total Charges : {:.2f}".format(round(mean_total,2)))

mode_total = data['TotalCharges'].mode()
print("The mode of Total Charges : {}".format(int(mode_total)))

median_total = data['TotalCharges'].median()
print("The median of Total Charges : {}".format(int(median_total)))

std_total = data['TotalCharges'].std()
print("The average of Total Charges : {:.2f}".format(std_total))

The mean of Total Charges : 2283.30
The mode of Total Charges : 20
The median of Total Charges : 1397
The average of Total Charges : 2266.77


Hasil perhitungan pada kolom total charges didapatkan berupa nilai rata-rata sebesar : 2283.30, modus : 20, median : 1397, dan standard deviasi : 2266.77

2) Pada kolom gender, Partner, Dependents,InternetService, Contract, PaymentMethod dan Churn lakukan perhitungan %proporsi terhadap Churn. Berikan rangkuman singkat tiap kolomnya.

In [ ]:
#Perhitungan pada kolom Proportion_gender
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['gender'].value_counts(normalize=True)
proportion = proportion.rename_axis('gender').reset_index()

proportion.head()

gender  Churn  Proportion(%)
0     No   5174            NaN
1    Yes   1869            NaN

Persentase kolom gender dengan churn adalah NaN (null) karena gender tidak memiliki value yang sama dengen churn.

In [ ]:
#Perhitungan pada kolom Proportion_Partner
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['Partner'].value_counts(normalize=True)
proportion = proportion.rename_axis('Partner').reset_index()

proportion.head()

Partner  Churn  Proportion(%)
0      No   5174       0.516967
1     Yes   1869       0.483033

Persentase kolom partner dengan churn untuk "No" adalah 0.516967 dan "Yes" 0.483033

In [ ]:
#Perhitungan pada kolom Proportion_Dependents
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['Dependents'].value_counts(normalize=True)
proportion = proportion.rename_axis('Dependents').reset_index()

proportion.head()

Dependents  Churn  Proportion(%)
0         No   5174       0.700412
1        Yes   1869       0.299588

Persentase kolom dependents dengan churn untuk "No" adalah 0.700412 dan "Yes" 0.299588

In [ ]:
#Perhitungan pada kolom Proportion_InternetService
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['InternetService'].value_counts(normalize=True)
proportion = proportion.rename_axis('InternetService').reset_index()

proportion.head()

InternetService  Churn  Proportion(%)
0              No   5174       0.216669
1             Yes   1869            NaN

Persentase kolom dependents dengan churn untuk "No" adalah 0.216669 dan tidaka ada value "Yes"

In [ ]:
#Perhitungan pada kolom Proportion_Contract
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['Contract'].value_counts(normalize=True)
proportion = proportion.rename_axis('Contract').reset_index()

proportion.head()

Contract  Churn  Proportion(%)
0       No   5174            NaN
1      Yes   1869            NaN

Persentase kolom Contract dengan churn adalah NaN (null) karena Contract tidak ada value yang sama dengen churn.

In [ ]:
#Perhitungan pada kolom Proportion_PaymentMethod
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['PaymentMethod'].value_counts(normalize=True)
proportion = proportion.rename_axis('PaymentMethod').reset_index()

proportion.head()

PaymentMethod  Churn  Proportion(%)
0            No   5174            NaN
1           Yes   1869            NaN

Persentase kolom PaymentMethod dengan churn adalah NaN (null) karena PaymentMethod tidak ada value yang sama dengen churn.

In [ ]:
#Perhitungan pada kolom Proportion_Churn
proportion = pd.DataFrame(data['Churn'].value_counts())
proportion['Proportion(%)'] = data['Churn'].value_counts(normalize=True)

proportion.head()

Churn  Proportion(%)
No    5174        0.73463
Yes   1869        0.26537

Persentase churn dengan value "No" adalah 0.73463 dan "Yes" 0.26537